src/Generating/FinishGen.h

```diff
@@ -70,6 +70,71 @@ class cFinishGenNetherClumpFoliage :
 
 
 
+class cFinishGenClumpTopBlock :
+	public cFinishGen
+{
+public:
+	// Contains the meta, type and weight for a clump block
+	struct BlockInfo
```

Comment:
Still, Foliage is the primary function, so it might be worth using it as the name.

The point is, when you later reference this as `block`, it creates a lot of confusion because it's unclear whether a `block` is this foliage info, or a topmost block in the chunk, or whatever else.